beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/performance/DefaultPerformanceTracker.java

```diff
@@ -140,7 +144,11 @@ public void onSlot(final UInt64 slot) {
       reportingTasks.add(reportSyncCommitteePerformance(currentEpoch));
     }
 
-    SafeFuture.allOf(reportingTasks.toArray(SafeFuture[]::new)).handleException(LOG::error).join();
+    final long startTime = System.currentTimeMillis();
```

Comment:
Just to be more precise I may add the start time before the line `final List<SafeFuture<?>> reportingTasks = new ArrayList<>();` to avoid some of the futures already starting or some additional logic before returning the futures.

Reply:
I don't think what you said is true. I'll argue when I'll be back on the desk.

Reply:
Ok so, future wont start until there is something asking for the result of it (which is ultimately the `join` down here).
But there there are also some synchronous calculations as you mentioned so make sense to start counting time earlier.